### PR TITLE
Fix vinylcontrol switching modes on track load

### PR DESCRIPTION
**Pull request by ywwg**
This feels a little hacky to me but it does work.

### CODE REVIEW

**Comment by ywwg** [comment]
fixes https://bugs.launchpad.net/mixxx/+bug/1438371

File: src/engine/cuecontrol.h [hunk 1]
```diff
@@ -153,6 +154,8 @@ class CueControl : public EngineControl {
     ControlPushButton* m_pCueGotoAndPlay;
     ControlPushButton* m_pCueGotoAndStop;
     ControlPushButton* m_pCuePreview;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
```
Comment (daschuer) [inline src/engine/cuecontrol.h]: Do not use ScopedPointers here, just parent the controls to CueControl.

**Comment by daschuer** [comment]
We need to make the engine buffer aware of positions behind the track to make this work reliable. 
The current position should also be preserved over an eject load cycle.

**Comment by daschuer** [comment]
This should also honor the track load behavior settings. If it is set to Load track at cue. It should do so. 
There are probably vinyl users that are confused with the new feature and this way, they can switch it off.

**Comment by ywwg** [comment]
> We need to make the engine buffer aware of positions behind the track to make this work reliable.
> The current position should also be preserved over an eject load cycle.

This already all works -- when a track is loaded the vinyl code performs a "force resync" which causes the playhead to seek to the current absolute position.

> load behavior settings.

done

**Comment by ywwg** [comment]
> positions behind the track

I don't understand what this means, but the feature seems to work correctly even if the user loads a track that is longer or shorter than the current absolute position.

**Comment by daschuer** [comment]
The following scenario does not work 
* play a long track to end  
* load a short track 
* position marker and track time is "seeked" track end 
* Turn the vinyl backward 
* Waveform is shaking due to every absolute position is clamped to the track end.
* The state is inconsistent vinyl and Mixxx do not match. 
* Expected behavior: Track time gives always a a feedback of the vinyl position.
(Dos not change due to loading a track) 
* Nice to have: Waveform is moving behind the track.

**Comment by ywwg** [comment]
The correct fix would be to allow positions past the end like we allow positions past the start.  We know that will work fine since the RAMAN knows how to process samples past the end.  I'll try adding a conditional to the stop-at-end code to not stop if vinyl control is on and absolute.

**Comment by ywwg** [comment]
(This would mean repeat mode would not work for vinyl / abs mode, but I'm ok with that.)

**Comment by ywwg** [comment]
Actually we don't support reading past the end yet, so it would be a more difficult change to fix that.  I'm not sure how big a problem it is to have the bug you describe, either way there is no playback, and it's pretty clear to the user what's going on -- the needle is too far down the record and they have to drop it earlier.

**Comment by daschuer** [comment]
OK this sounds like a not reasonable work to fix it yet. 
Does it still switch to relative mode when repeat is on? I think it should.
If yes, write a bug about the waveform shaking and merge it, since it fixes the original bug.

**Comment by ywwg** [comment]
Yes, repeat mode still works as expected.  Bug filed: https://bugs.launchpad.net/mixxx/+bug/1440751

File: src/engine/cuecontrol.h (hunk 1 above)
Comment (ywwg) [inline src/engine/cuecontrol.h]: I'm going to just keep the style consistent in this file and use a bare pointer

**Pull request merged**